internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -701,40 +735,50 @@ func (r *ClusterDeploymentsReconciler) SetupWithManager(mgr ctrl.Manager) error
 		return reply
 	}
 
+	// Reconcile the ClusterDeployment referenced by this AgentClusterInstall.
+	mapClusterInstallToClusterDeployment := func(a client.Object) []reconcile.Request {
+		aci, ok := a.(*hiveext.AgentClusterInstall)
+		if !ok {
+			r.Log.Errorf("%v was not an AgentClusterInstall", a) // shouldn't be possible
+			return []reconcile.Request{}
+		}
+		return []reconcile.Request{
+			{
+				NamespacedName: types.NamespacedName{
+					Namespace: aci.Namespace,
+					Name:      aci.Spec.ClusterDeploymentRef.Name,
+				},
+			},
+		}
+	}
+
 	clusterDeploymentUpdates := r.CRDEventsHandler.GetClusterDeploymentUpdates()
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&hivev1.ClusterDeployment{}).
 		Watches(&source.Kind{Type: &corev1.Secret{}}, handler.EnqueueRequestsFromMapFunc(mapSecretToClusterDeployment)).
+		Watches(&source.Kind{Type: &hiveext.AgentClusterInstall{}}, handler.EnqueueRequestsFromMapFunc(mapClusterInstallToClusterDeployment)).
 		Watches(&source.Channel{Source: clusterDeploymentUpdates}, &handler.EnqueueRequestForObject{}).
```

Comment:
If desired you could do another mapfunc here and skip if the clusterInstallRef doesn't match an AgentClusterInstall, then you're never Reconciling a ClusterDeployment you don't care about.

Reply:
+1

Reply:
What about garbage collection? if the user disconnect the agent cluster install and doesn't delete it?

Reply:
Garbage collection will be done on follow up  PRs. I opened a JIRA MGMT-6039